aten/CMakeLists.txt

```diff
@@ -65,7 +65,15 @@ ENDIF()
 IF(MSVC)
   # we want to respect the standard, and we are bored of those **** .
   ADD_DEFINITIONS(-D_CRT_SECURE_NO_DEPRECATE=1)
-  LIST(APPEND CUDA_NVCC_FLAGS "-Xcompiler /wd4819 -Xcompiler /wd4503 -Xcompiler /wd4190 -Xcompiler /wd4244 -Xcompiler /wd4251 -Xcompiler /wd4275 -Xcompiler /wd4522")
+  LIST(APPEND CUDA_NVCC_FLAGS
+        -Xcompiler /wd4190 # identifier1' has C-linkage specified, but returns UDT 'identifier2' which is incompatible with C
```

Comment:
Looks nice :)

Do we want to also directly add them to all compilation units in cmake/MiscCheck.cmake?

Reply:
So... I didn't actually want to be working on this, I was trying to write some bootcamp tasks for people :)

I just updated this patch, there seem to be four places where we separately set warnings. Not sure if we have a planned code structure for dealing with this.

Reply:
Based on the current organization of code, we should probably move things to the cmake/ folder somewhere like cmake/Platform.cmake but this can be done separately - feel free to land the code.